masp_primitives/src/transaction/txid.rs

```diff
@@ -0,0 +1,365 @@
+use std::borrow::Borrow;
```

Comment:
Since the binding signature is no longer over `expiry_height, is there a problem where an unexecuted masp tx can be repackaged in a different outer tx and resubmitted?

Reply:
I was also considering reintroducing the `zfuture` code. In the interest of making the diff as small as possible

Reply:
> Since the binding signature is no longer over `expiry_height, is there a problem where an unexecuted masp tx can be repackaged in a different outer tx and resubmitted?

There shouldn't be an issue with repackaging attacks because the hash of the Namada `Transfer` target address is used as the public key of a transparent output that is then signed over in the MASP `Transaction`. The Namada validity predicate checks for this.

More generally, because the Namada VP checks for consistency between the inner `Transaction` and outer `Transfer`, and because the inner `Transaction` is signed over, it should be hard to do any repackaging. See https://github.com/anoma/namada/blob/main/documentation/specs/src/masp/ledger-integration.md#boundary-conditions .

Reply:
> I was also considering reintroducing the `zfuture` code. In the interest of making the diff as small as possible

That sounds fine. I'd need to study the `zfuture` code to understand what exactly it does...